### PR TITLE
Bluetooth: Shell: Various improvements to gatt commands

### PR DESCRIPTION
**Pull request by Vudentz**
This adds commands to get/set local attribute values, and improves show command which can now print the database memory usage.

### CODE REVIEW

File: subsys/shell/shell.c [hunk 1]
```diff
@@ -1355,6 +1355,28 @@ void shell_fprintf(const struct shell *shell, enum shell_vt100_color color,
 	k_mutex_unlock(&shell->ctx->wr_mtx);
 }
 
+void shell_hexdump(const struct shell *shell, const u8_t *data, size_t len)
+{
+	int n = 0;
+
+	while (len--) {
+		if (n % 16 == 0) {
+			shell_fprintf(shell, SHELL_NORMAL, "%08X: ", n);
```
Comment (nordic-krch) [inline subsys/shell/shell.c]: why not use `shell_print` like below?

File: subsys/bluetooth/shell/gatt.c [hunk 2]
```diff
@@ -764,6 +833,98 @@ static int cmd_metrics(const struct shell *shell, size_t argc, char *argv[])
...
+
+	data.shell = shell;
+	data.argc = argc - 2;
+	data.argv = argv + 2;
+	data.err = 0;
+
+	bt_gatt_foreach_attr(handle, handle, set_cb, &data);
+
+	if (data.err < 0) {
+		return -ENOEXEC;
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/gatt.c]: you can consider adding `shell_error` with an error information.

File: subsys/bluetooth/shell/gatt.c [hunk 3]
```diff
@@ -478,19 +478,88 @@ static int cmd_unsubscribe(const struct shell *shell,
...
+		data.uuid.uuid.type = BT_UUID_TYPE_16;
+		data.uuid.val = strtoul(argv[1], NULL, 16);
+	} else {
+		data.uuid.val = 0;
+	}
+
+	bt_gatt_foreach_attr(0x0001, 0xffff, print_attr, &data);
+
+	if (!stats.attr_count) {
+		return 0;
```
Comment (jarz-nordic) [inline subsys/bluetooth/shell/gatt.c]: Maybe some feedback cound be printed here?

**Comment by jarz-nordic** [comment]
Great idea adding shell_hexdump.

File: subsys/bluetooth/shell/gatt.c (hunk 2 above)
Comment (Vudentz) [reply subsys/bluetooth/shell/gatt.c]: There are error printed on set_cb.

File: subsys/bluetooth/shell/gatt.c (hunk 3 above)
Comment (Vudentz) [reply subsys/bluetooth/shell/gatt.c]: I will add a message when no attribute are found.

**Review by jakub-uC — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**